### ISSUE TITLE
Generated TRecord.from(IT) method behaves differently from Record.from() with respect to NOT NULL DEFAULT columns

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14727]
Given:

```sql
CREATE TABLE t_defaults (
  e1 INTEGER,
  e2 INTEGER NOT NULL,
  e3 INTEGER          DEFAULT 1,
  e4 INTEGER NOT NULL DEFAULT 1
);
```

With #2700 and #3582, we've ensured that when calling

```java
TDefaultsRecord r = new TDefaultsRecord();
r.from(new TDefaults()); // TDefaults being a POJO
```

We'd avoid setting `e2` and `e4` to `changed` and `null`.

But when `<interfaces>true</interfaces>` is specified, the behaviour is different. We're now setting column values individually for performance reasons. The generated method does this:

```java
    @Override
    public void from(ITDefaults from) {
        setE1(from.getE1());
        setE2(from.getE2());
        setE3(from.getE3());
        setE4(from.getE4());
    }
```

It skips the call to `Tools::resetChangedOnNotNull` from `AbstractRecord::from0`:

```java
    private final void from0(Object source, int[] targetIndexMapping) {
        if (source == null) return;

        // [#2520] TODO: Benchmark this from() method. There's probably a better implementation
        from(
            Tools.configuration(this).recordUnmapperProvider().provide(source.getClass(), (FieldsImpl) fields.fields).unmap(prepareArrayOrIterableForUnmap(source, targetIndexMapping)),
            targetIndexMapping
        );

        // [#2700] [#3582] If a POJO attribute is NULL, but the column is NOT NULL
        // then we should let the database apply DEFAULT values
        resetChangedOnNotNull(this);
    }
```

----
As reported here:
- https://stackoverflow.com/q/75602163/521799

**Comment by lukaseder** [comment]
This is a subtly incompatible change. In rare cases, users may rely on the `NULL` value being passed to the database for a trigger to pick up the intended change and apply something else.

However, the inconsistency of `from(Object)` with `from(ITDefaults)` is more significant. The two methods should not expose any different behaviour at all, given that `ITDefaults` can always be upcast to `Object`: Liskov principle

**Comment by lukaseder** [comment]
Implemented for jOOQ 3.18.0
